akka-http-tests/src/test/java/akka/http/javadsl/server/JavaRouteTest.java

```diff
@@ -0,0 +1,351 @@
+package akka.http.javadsl.server;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.junit.Test;
+
+import akka.http.javadsl.model.ContentType;
+import akka.http.javadsl.model.ContentTypes;
+import akka.http.javadsl.model.HttpCharsets;
+import akka.http.javadsl.model.HttpEntity;
+import akka.http.javadsl.model.HttpRequest;
+import akka.http.javadsl.model.HttpResponse;
+import akka.http.javadsl.model.MediaRanges;
+import akka.http.javadsl.model.MediaTypes;
+import akka.http.javadsl.model.RequestEntity;
+import akka.http.javadsl.model.StatusCodes;
+import akka.http.javadsl.model.headers.Accept;
+import akka.http.javadsl.model.headers.RawHeader;
+import akka.http.javadsl.testkit.JUnitRouteTest;
+//FIXME discuss how to provide a javadsl.CustomHeader where render() is either pre-implemented or trivial to write in Java
+import akka.http.scaladsl.model.headers.CustomHeader;
+import akka.japi.pf.PFBuilder;
+import akka.util.ByteString;
+import static akka.http.javadsl.server.StringUnmarshaller.*;
+import static akka.http.javadsl.server.PathMatchers.*;
```

Comment:
Extending a class is an OK way to get started and for quick docs and tests, but my experience with static imports is actually the opposite of what is being stated. Getting to constants by extending an interface/base class is also explicitly mentioned as an [anti-pattern](https://docs.oracle.com/javase/1.5.0/docs/guide/language/static-import.html) even by Sun. Remember, most of `StringUnmarshallers` and `PathMatchers` will become "just" `public static final`s.

 Having to extend a class to get to constants has two big drawbacks:
- A class can only extend 1 base class. Requiring this makes it impossible to mix akka with other frameworks.
- All of the inherited constants appear on the public API of the class being written (unless we make them protected, completely removing the ability to statically import)

I'll refactor the aforementioned classes into constants first, let's see what it looks like then. Currently people also don't need to extend `StatusCodes` to get to the HTTP status codes either.